omnigibson/prims/entity_prim.py

```diff
@@ -642,27 +575,15 @@ def set_joint_efforts(self, efforts, indices=None, normalized=False):
             normalized (bool): Whether the inputted joint efforts should be interpreted as normalized values. Default
                 is False
         """
-        # Run sanity checks -- make sure our handle is initialized and that we are articulated
-        assert self._handle is not None, "handles are not initialized yet!"
+        # Run sanity checks -- make sure we are articulated
         assert self.n_joints > 0, "Tried to call method not intended for entity prim with no joints!"
 
         # Possibly de-normalize the inputs
         if normalized:
             efforts = self._denormalize_efforts(efforts=efforts, indices=indices)
 
-        # Grab current DOF states
-        dof_states = self._dc.get_articulation_dof_states(self._handle, _dynamic_control.STATE_EFFORT)
-
-        # Possibly set specific values in the array if indies are specified
-        if indices is None:
-            new_efforts = efforts
-        else:
-            new_efforts = dof_states["effort"]
-            new_efforts[indices] = efforts
-
         # Set the DOF states
-        dof_states["effort"] = new_efforts
-        self._dc.set_articulation_dof_states(self._handle, dof_states, _dynamic_control.STATE_EFFORT)
+        self._articulation_view.set_joint_efforts(velocities=efforts, joint_indices=indices)
```

Comment:
a couple Qs:

- shouldn't it be `efforts=...`, not `velocities=...`?
- so this can now handle variable-length arrays now, based on `joint_indices=...`?

Reply:
Yup, fixed. Thanks!